nupic/research/monitor_mixin/plot.py

```diff
@@ -46,40 +49,70 @@ def __init__(self, monitor, title):
     plt.show()
 
 
+
+  def _initFigure(self):
+    fig = plt.figure()
+    fig.suptitle(self._prettyPrintTitle())
+    return fig
+
+
+
+  def _prettyPrintTitle(self):
+    if self._monitor.mmName is not None:
+      return "[{0}] {1}".format(self._monitor.mmName, self._title)
+    return self._title
+
+
+
   def addGraph(self, data, position=111, xlabel=None, ylabel=None):
     ax = self._addBase(position, xlabel=xlabel, ylabel=ylabel)
     ax.plot(data)
     plt.draw()
 
 
-  def addHistogram(self, data, position=111, xlabel=None, ylabel=None,
-                   bins=None):
+
+  def addImage(self, data, position=111, xlabel=None, ylabel=None,
+               cmap=colorModel.Greys, aspect="auto", interpolation="nearest"):
     """
-    @param bucketSize (int) Size of each bucket
+    Adds an image to the plot's figure.
+    :param data: a 2D array
```

Comment:
See the python doxygen syntax example [here](https://github.com/numenta/nupic/wiki/Documenting-NuPIC-with-Doxygen#example).